examples/src/utils/loaders.ts

```diff
@@ -41,13 +51,15 @@ export function createModelIdentifierFromUrlParams(
  * @param model Model identifier. If a number, the model is assumed to be stored in CDF.
  * @param project If model is a number (i.e. model ID), project must be provided.
  */
-export async function loadCadModelFromCdfOrUrl(model: ModelIdentifier): Promise<reveal.CadModel> {
+export async function loadCadModelFromCdfOrUrl(model: ModelIdentifier, client?: CogniteClient): Promise<reveal.CadModel> {
   if (isUrlModelIdentifier(model)) {
     return reveal.loadCadModelByUrl(model.modelUrl);
   } else {
-    const client = new CogniteClient({ appId: 'cognite.reveal.example' });
-    client.loginWithOAuth({ project: model.project });
-    await client.authenticate();
+    if (!client) {
+      client = new CogniteClient({ appId: 'cognite.reveal.example' });
```

Comment:
Food for thought: newing things inside functions / classes often runs the risk of the user creating said elements multiple times in code. Say you call loadCadModelFromCdfOrUrl without providing a client you technically create a client each time the user calls on said method. I don't like that, what are your thoughts on the matter?

Reply:
I don't care tbh. It's an example. The reason it's like this is because the client doesn't need to be initialized always and this enabled to not copy code in each example.